src/calendar.ts

```diff
@@ -0,0 +1,193 @@
+    // NOTE: We can't go upwards by using `Interactor().find(...)`
+    const nextMonthElement = getHeaderElement(element)?.lastElementChild;
+    if (isHTMLElement(nextMonthElement)) nextMonthElement.click();
+  });
+}
+function goToPrevMonth({ perform }: Interactor<HTMLElement, any>) {
+  return perform((element) => {
+    // NOTE: We can't go upwards by using `Interactor().find(...)`
+    const prevMonthElement = getHeaderElement(element)?.firstElementChild;
+    if (isHTMLElement(prevMonthElement)) prevMonthElement.click();
+  });
+}
+
+async function goToYear(interactor: Interactor<HTMLElement, any>, targetYear: number) {
+  let currentMonth = await getCurrentMonth(interactor);
+  let currentYear = await getCurrentYear(interactor);
+
+  if (!currentMonth || !currentYear) throw new Error("Can't get current month and year");
+  if (currentYear == targetYear) return;
+  const step = currentYear < targetYear ? () => goToNextMonth(interactor) : () => goToPrevMonth(interactor);
+  const targetMonth = currentMonth;
+  while (currentYear != targetYear || currentMonth != targetMonth) {
+    await step();
+    await delay(1000);
+    const prevMonth: string | undefined = currentMonth;
+    currentMonth = await getCurrentMonth(interactor);
+    currentYear = await getCurrentYear(interactor);
+    if (prevMonth == currentMonth)
+      throw new Error(
+        `Can't set '${targetYear}' year. It might happened because of 'minDate/maxDate' or 'disableFuture/disablePast' props`
+      );
+  }
+}
+async function goToMonth(
+  interactor: Interactor<HTMLElement, any>,
+  targetMonth: string,
+  directionStep: () => Interaction<void>,
+  currentYear?: number
+) {
+  let currentMonth = await getCurrentMonth(interactor);
+
+  if (!currentMonth || !currentYear) throw new Error("Can't get current month and year");
+  if (currentMonth == targetMonth) return;
+
+  const targetYear = currentYear;
+  while (currentYear != targetYear || currentMonth != targetMonth) {
+    await directionStep();
+    await delay(1000);
+    const prevMonth: string | undefined = currentMonth;
+    currentMonth = await getCurrentMonth(interactor);
+    currentYear = await getCurrentYear(interactor);
+    if (currentYear != targetYear || currentMonth == prevMonth)
+      throw new Error(
+        `Can't set '${targetMonth}' month. It might happened because of 'minDate/maxDate' or 'disableFuture/disablePast' props`
+      );
+  }
+}
+async function goToDay(interactor: Interactor<HTMLElement, any>, day: number) {
+  // NOTE: We can't find day if user has custom day render
+  const dayInteractor = interactor.find(HTML.selector(".MuiPickersCalendar-week > [role='presentation']")(String(day)));
+  try {
+    await dayInteractor.has({ className: not(including("MuiPickersDay-dayDisabled")) });
+  } catch (_) {
+    throw new Error(`Can't select ${day} day because it's disabled`);
+  }
+  return dayInteractor.click();
+}
+
+export const getCalendar = (utils: ReturnType<typeof useUtils>) =>
```

Comment:
```suggestion
export const createCalendar = (utils: ReturnType<typeof useUtils>) =>
```

Also, let's explicitly define our own type here so that development of the test suite is not coupled to the typings of `@material-ui/pickers`.